monitors/ContractMonitor.js

```diff
@@ -0,0 +1,173 @@
+const { Logger } = require("../financial-templates-lib/logger/Logger");
+
+const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
+const networkUtils = require("../common/PublicNetworks");
+
+class ContractMonitor {
+  constructor(expiringMultiPartyEventClient, account, umaLiquidatorAddress, umaDisputerAddress) {
+    // Bot and ecosystem accounts.
+    this.account = account;
+    this.umaLiquidatorAddress = umaLiquidatorAddress;
+    this.umaDisputerAddress = umaDisputerAddress;
```

Comment:
A few comments:
1. nit: since we're open sourcing this with the intention of encouraging others to use these bots, it might make sense to change the names of these variables. Something like `monitoredLiquidator`, `watchedLiquidator`, or something similar.
2. This might be worthy of a TODO if it's complicated to add, but it might make sense to make these arrays rather than single addresses to make the monitor bot a bit more flexible. Thoughts?

Reply:
1. agreed and done.
2. great idea. we might have multiple bots. I take in an array in wallet monitor but here placed the restriction to having one liquidator and one dispute bot. I've refactored this to enable an array of liquidators and disruptors to be fed in and checked in the script.

Follow up question is if we want to remove any uma related text how should I handel the bot printing things like this:
```
0xB3c...78206 (UMA liquidator bot) initiated liquidation for...
```
When this says "UMA liquidator bot". We could set this as an env variable or pass it in but both of these options are a bit clunky.

Reply:
You can check with @allisonlu22, but I don't think it would be that much worse if we said:
```
0xB3c...78206 (Watched liquidator bot) initiated liquidation for...
```

The other option would be to pass in some sort of label for each liquidation/dispute bot along with the addresses, but that may be overkill.